kubewarden_test.go

```diff
@@ -0,0 +1,339 @@
+package sdk
+
+import (
+	"testing"
+
+	appsv1 "github.com/kubewarden/k8s-objects/api/apps/v1"
+	batchv1 "github.com/kubewarden/k8s-objects/api/batch/v1"
+	corev1 "github.com/kubewarden/k8s-objects/api/core/v1"
+	"github.com/kubewarden/policy-sdk-go/protocol"
+	"github.com/mailru/easyjson"
+	"github.com/mailru/easyjson/jwriter"
+)
+
+func CreateValidationRequest(object easyjson.Marshaler, kind string) (protocol.ValidationRequest, error) {
+	w := jwriter.Writer{}
+	object.MarshalEasyJSON(&w)
+	value, err := w.BuildBytes()
+	if err != nil {
+		return protocol.ValidationRequest{}, err
+	}
+
+	validationRequest := protocol.ValidationRequest{
+		Settings: easyjson.RawMessage{},
+		Request: protocol.KubernetesAdmissionRequest{
+			Kind: protocol.GroupVersionKind{
+				Kind: kind,
+			},
+			Object: value,
+		},
+	}
+
+	return validationRequest, nil
+}
+
+func CheckIfAutomountServiceAccountTokenIsTrue(t *testing.T, rawResponse []byte) protocol.ValidationResponse {
+	response := protocol.ValidationResponse{}
+	if err := easyjson.Unmarshal(rawResponse, &response); err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+
+	if !response.Accepted {
+		t.Fatalf("Response not accepted")
+	}
+
+	if len(response.MutatedObject.(map[string]interface{})) == 0 {
+		t.Fatalf("Request should be mutated")
+	}
+
+	return response
+}
```

Comment:
The name of this helper function confused me a bit. I also don't like to have helper functions take `testing.T` and deal with `t.Fatalf`. I prefer them to just return an error, and then let the unit test decide what to do.

I would rewrite this helper like that:

```go
// Build a ValidationResponse object. Returns an error if the validation response is not accepted
// and the incoming object has not been mutated
func CheckIfAutomountServiceAccountTokenIsTrue(t *testing.T, rawResponse []byte) (protocol.ValidationResponse, error) {
	response := protocol.ValidationResponse{}
	if err := easyjson.Unmarshal(rawResponse, &response); err != nil {
		return response, err
	}

	if !response.Accepted {
		return response, fmt.Errorf("Response not accepted")
	}

        // I **think** this should work too
	if response.MutatedObject == nil {
		return response, fmt.Errorf("Request not mutated")
	}

	return response
}
```